drivers/platform/x86/thinkpad-wmi.c

```diff
@@ -477,11 +517,19 @@ static ssize_t store_setting(struct device *dev,
 	struct thinkpad_wmi *thinkpad = dev_get_drvdata(dev);
 	struct dev_ext_attribute *ea = to_ext_attr(attr);
 	int item_idx = (uintptr_t)ea->var;
-	const char *item = thinkpad->settings[item_idx];
+	char *item = thinkpad->settings[item_idx];
 	int ret;
 	size_t buffer_size;
 	char *buffer;
 
+        int spleng=0;
+        int num=0;
+        spleng=strlen(item);
```

Comment:
can you add a comment here ?

also would it make sense to use strnlen instead ?